sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/CallToAsyncMethodShouldNotBeBlocking.cs

```diff
@@ -117,6 +121,25 @@ private static void ReportOnViolation(SyntaxNodeAnalysisContext context)
                 messageArgs: MemberNameToMessageArguments[memberAccessNameName]));
         }
 
+        private static bool IsResultInContinueWithCall(string memberAccessName, MemberAccessExpressionSyntax memberAccess) =>
+            memberAccessName == ResultName &&
+            memberAccess.Expression is IdentifierNameSyntax identifierNameSyntax &&
+            identifierNameSyntax.GetName() is {} identifierName &&
+            memberAccess.FirstAncestorOrSelf<InvocationExpressionSyntax>(invocation => IsContinueWithCallWithArgumentName(invocation, identifierName)) != null;
+
+        private static bool IsContinueWithCallWithArgumentName(InvocationExpressionSyntax invocation, string argumentName) =>
+            invocation.Expression.NameIs(ContinueWithName) &&
+            invocation.ArgumentList.Arguments.Any(argument => IsLambdaExpressionWithArgumentName(argument.Expression, argumentName));
+
+        private static bool IsLambdaExpressionWithArgumentName(ExpressionSyntax expression, string argumentName) =>
+            expression switch
+            {
+                SimpleLambdaExpressionSyntax lambda => lambda.Parameter.Identifier.ValueText == argumentName,
```

Comment:
_Minor:_ I would rename `lambda` to `simpleLambda` for consistency.

Reply:
In this case it will just make the line longer. We have all the context in the same line.